File: 1657. Determine if Two Strings Are Close/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if (word1.size() != word2.size()) {
            return false;
        }

        int freq_1[26] = {0};
        int freq_2[26] = {0};

        for (char s : word1) {freq_1[s-'a']++;}
        for (char s : word2) {freq_2[s-'a']++;}
       
                
        for (int i = 0; i < 26; i++) {
            if (freq_1[i] == 0 && freq_2[i] != 0) {
                return false;
            }
            if (freq_2[i] == 0 && freq_1[i] != 0) {
                return false;
            }
        }

        sort(freq_1, freq_1+26);
        sort(freq_2, freq_2+26);

        for(int i=0; i<26; i++){
            if(freq_1[i]!=freq_2[i]){
                return false;
            }
        }

        return true;
    }
};

File: 1657. Determine if Two Strings Are Close/Approach03.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False

        freq_1 = [0] * 26
        freq_2 = [0] * 26

        for s in word1:
            freq_1[ord(s) - ord('a')] += 1

        for s in word2:
            freq_2[ord(s) - ord('a')] += 1

        for i in range(26):
            if freq_1[i] == 0 and freq_2[i] != 0:
                return False
            if freq_2[i] == 0 and freq_1[i] != 0:
                return False

        freq_1.sort()
        freq_2.sort()

        for i in range(26):
            if freq_1[i] != freq_2[i]:
                return False

        return True
